lib/beaker/hypervisor/fusion.rb

```diff
@@ -54,6 +55,25 @@ def provision
         end
         time = Time.now - start
         @logger.notify "Spent %.2f seconds resuming VM" % time
+
+        begin
+          try_ssh_connection(host)
+        rescue *Beaker::SshConnection::RETRYABLE_EXCEPTIONS => e
+          @logger.notify "Obtaining IP information for #{host.name}"
+          ip = vm.network_info.data[vm.network_info.data.keys[0]]['ip_address']
+          if not ip
+            raise "Unable to connect to vm via hostname #{host.name}, and ip address is unavailable via vmname #{host[:vmname]}."
+          else
+            @logger.notify "Found IP address #{ip} for #{host.name}. Setting host.ip value"
+            host[:ip] = ip
+          end
```

Comment:
The assumption here is that the ip will work, considering that we are testing the ssh connection by name we should probably do the same for ip and then raise as invalid before doing additional work on the host.